app/src/main/java/org/jellyfin/androidtv/auth/ServerRepository.kt

```diff
@@ -92,6 +92,8 @@ class ServerRepositoryImpl(
 			.forEach { user ->
 				if (users.none { it.id == user.id })
 					users.add(user)
+				else
+					users[users.indexOfFirst { it.id == user.id }] = user
```

Comment:
```suggestion
				val index = users.indexOfFirst { it.id == user.id }
				if (index == -1) users.add(user)
				else users[index] = user
```
This way it will not search the list twice, although one problem with this behavior is that public users will replace private users, which uses a different class and might contain additional data in the future. I'm not sure if that's something we want to do.